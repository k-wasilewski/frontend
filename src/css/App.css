.App {
  background-color: lavender;
  min-height: 100vh;
  font-family: Lato
}

.menu {
  top: 5px;
  right: 10vw;
  background-color: #d3d3e6;
  box-shadow: 0 0 10px 10px #d3d3e6;
  min-height: 5px;
  min-width: 20px;
  float: left;
  position: fixed;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.menu-item {
  font-family: Lato;
  background-color: transparent;
  border-radius: 5px;
  display: inline-block;
  cursor: pointer;
  color: black;
  font-size: 20px;
  text-decoration: none;
  margin: 5px;
}

.menu-item-div:hover {
  font-weight: bold;
}

#menu-btn {
  cursor: pointer;
  height: 60px;
  float: right;
}

.main {
  padding-top: 10vh;
  clear: both;
  padding-left: 15%;
}

.hidden {
  display: none;
}

.form > .col1 {
  width: 15%;
  float: left;
  display: block;
}

.form > .col2 {
  float: left;
  width: 30%;
}

.col1 > p {
  height: 1.5em;
}

.col1 > button {
  margin-top: 40%;
}

.col2 > p {
  height: 1.5em;
}

.form > h2 {
  padding-bottom: 5%;
}