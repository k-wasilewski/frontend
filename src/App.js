import React, {Component} from 'react';
import './css/App.css';
import Router from "./Router";
import './css/App.css';
import Menu from './components/Menu'

class App extends Component {
  render() {
    return (
        <div className="App">
            <Menu />
            <Router />
        </div>
    );
  }
}

export default App;
