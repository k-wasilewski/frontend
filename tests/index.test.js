import React from 'react';
import store from "../src/redux/store";
import {BrowserRouter} from "react-router-dom";
import App from "../src/App";
import {Provider} from "react-redux";
import {render} from "react-dom";

describe("index rendering specification", () => {
    const root = document.createElement('div');

    beforeEach(() => {
        document.body.appendChild(root);

        render(
            <Provider store={store}>
                <BrowserRouter>
                    <App />
                </BrowserRouter>
            </Provider>,
            root
        );
    });

    it('index is rendered to "root" div', () => {
        expect(root.children[0].classList).toContain('App');
    });

    it('index is rendered with divs "menu" and "main"', () => {
        const menu = document.body.children[0].children[0].children[0].children[0];
        const main = document.body.children[0].children[0].children[1].children[0];

        expect(Array.from(menu.classList)).toContain('menu');
        expect(Array.from(main.classList)).toContain('main');
    });
});