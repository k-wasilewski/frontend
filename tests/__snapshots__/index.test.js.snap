// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index rendering specification index is rendered 1`] = `
<div
  className="App"
>
  <header>
    <div
      id="user"
    >
      <br />
    </div>
    <div
      className="menu hidden"
    >
      <nav>
        <a
          aria-current="page"
          className="menu-item active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          <button
            className="menu-item"
          >
            <div
              className="menu-item-div"
            >
              Nowe zamówienie
            </div>
          </button>
        </a>
        <a
          aria-current={null}
          className="menu-item"
          href="/login"
          onClick={[Function]}
        >
          <button
            className="menu-item"
          >
            <div
              className="menu-item-div"
            >
              Logowanie
            </div>
          </button>
        </a>
      </nav>
    </div>
    <img
      alt="menu button"
      id="menu-btn"
      onClick={[Function]}
      src={Object {}}
    />
  </header>
  <div
    className="router"
  >
    <div
      className="main"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="form"
        >
          <p
            className="resp"
          />
          <h2>
            Nowe zamówienie
          </h2>
          <p
            className="hidden invalid"
          >
            
          </p>
          <div
            className="col1"
          >
            <p>
              Imię:
            </p>
            <p>
              Wiek:
            </p>
            <button>
              Dalej
            </button>
          </div>
          <div
            className="col2"
          >
            <p>
              <input
                id="nameInput"
                name="name"
                onChange={[Function]}
                type="text"
                value=""
              />
            </p>
            <p>
              <input
                id="ageInput"
                name="age"
                onChange={[Function]}
                type="number"
                value=""
              />
            </p>
          </div>
        </div>
      </form>
    </div>
    <p
      id="copyright"
    >
       © Kuba Wasilewski, 2020 
    </p>
  </div>
</div>
`;
